### PR TITLE
refactor(ui): refactor native supported social connectors jsbridge

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Refactor native supported social connectors jsbridge api schema
black font
White font

```
supportedConnector: {
  universal: Boolean;
  nativeTargets: string[]
}
```

## Linear Issue Reference

log-2571
log-2563

## Testing

test locally
UT case updated
@logto-io/eng  @xiaoyijun  @gao-sun

### CODE REVIEW

File: packages/ui/src/hooks/utils.ts [hunk 1]
```diff
@@ -90,27 +92,33 @@ export const filterSocialConnectors = (socialConnectors?: ConnectorData[]) => {
     return Array.from(connectorMap.values());
   }
 
+  /**
+   * Native Webview Environment
+   * Accepts both Native and universal platform connectors.
+   * Insert universal connectors only if there is no  Native platform connector provided with the same target.
```
Comment (gao-sun) [inline packages/ui/src/hooks/utils.ts:98]:
```suggestion
   * Insert universal connectors only if there is no native platform connector provided with the same target.
```

File: packages/ui/src/hooks/utils.ts [hunk 2]
```diff
@@ -90,27 +92,33 @@ export const filterSocialConnectors = (socialConnectors?: ConnectorData[]) => {
     return Array.from(connectorMap.values());
   }
 
+  /**
+   * Native Webview Environment
+   * Accepts both Native and universal platform connectors.
```
Comment (gao-sun) [inline packages/ui/src/hooks/utils.ts:97]:
```suggestion
   * Accepts both native and universal platform connectors.
```

**Review by gao-sun — APPROVED** [review]

File: packages/ui/src/hooks/utils.ts (hunk 1 above)
Comment (simeng-li) [reply packages/ui/src/hooks/utils.ts]: updated

File: packages/ui/src/hooks/utils.ts (hunk 2 above)
Comment (simeng-li) [reply packages/ui/src/hooks/utils.ts]: updated

**Pull request merged**